packages/freezed/test/common_types_test.dart

```diff
@@ -0,0 +1,231 @@
+      var value = CommonSuperSubtype('a', 1337, 42.24);
+      expect(value.a, 'a');
+      expect(value.b, 1337);
+      expect(value.c, 42.24);
+
+      value = CommonSuperSubtype.named('b', 12.34, 56.78);
+      expect(value.a, 'b');
+      expect(value.b, 12.34);
+      expect(value.c, 56.78);
+    });
+
+    test('copy has all with common subtypes available', () {
+      var value = CommonSuperSubtype('a', 1337, 42);
+      expect(value.a, 'a');
+
+      value = value.copyWith(a: 'b', c: 24.42);
+      expect(value.a, 'b');
+      expect(value.c, 24.42);
+    });
+
+    test(
+      'copy has shared params with common subtypes available',
+      () async {
+        await expectLater(compile(r'''
+import 'common_types.dart';
+
+void main() {
+  final param = CommonSuperSubtype('a', 1337, 42);
+  param.copyWith(a: '2');
+}
+'''), completes);
+
+        await expectLater(compile(r'''
+import 'common_types.dart';
+
+void main() {
+  final param = CommonSuperSubtype('a', 1337, 42);
+  param.copyWith(c: 12.34);
+}
+'''), completes);
+
+        await expectLater(compile(r'''
+import 'common_types.dart';
+
+void main() {
+  final param = CommonSuperSubtype('a', 1337, 42);
+  param.copyWith(c: 24.42);
+}
+'''), completes);
+
+        await expectLater(compile(r'''
+import 'common_types.dart';
+
+void main() {
+  final param = CommonSuperSubtype('a', 1337, 42);
+  param.copyWith(a: null);
+}
+'''), throwsCompileError);
+
+        await expectLater(compile(r'''
+import 'common_types.dart';
+
+void main() {
+  final param = CommonSuperSubtype('a', 1337, 42);
+  param.copyWith(c: null);
+}
+'''), throwsCompileError);
+      },
+    );
+
+    test(
+      'copy does not have shared params without common subtype available',
+      () async {
+        await expectLater(compile(r'''
+import 'common_types.dart';
+
+void main() {
+  final param = CommonSuperSubtype('a', 1337, 42);
+  param.copyWith(b: 42);
+}
+'''), throwsCompileError);
+
+        await expectLater(compile(r'''
+import 'common_types.dart';
+
+void main() {
+  final param = CommonSuperSubtype('a', 1337, 42);
+  param.copyWith(b: null);
+}
+'''), throwsCompileError);
+      },
+    );
+  });
+
+  group('GeneratedDependency', () {
+    test('should have getters available', () async {
+      final param = GeneratedDependency(Dependency0());
+      expect(param.a, isA<Dependency0>());
+    });
+  });
+
+  group('CommonInterfaceSupertype', () {
+    test('has common param available', () {
+      var value = CommonInterfaceSupertype(CommonInterfaceB());
+      expect(value.param, isA<CommonInterfaceB>());
+    });
+
+    test('does not have copy available', () async {
+      await expectLater(compile(r'''
+import 'common_types.dart';
+
+void main() {
+  CommonInterfaceSupertype param =
+    CommonInterfaceSupertype(CommonInterfaceB());
+  param.copyWith;
+}
+'''), throwsCompileError);
+    });
+  });
+
+  group('CommonTypedefs', () {
+    test('generates correct typedefs', () async {
```

Comment:
What is this test trying to do?

In general I don't like the idea of analyzing the source code like that. Chances are there's a better way to do this that doesn't rely on the super complex analyzer API

Reply:
I literally just copied the other typedef test and adapted it here. If you don't like it like that I can rewrite it to be like the other tests